tools/upgrade-deps/src/npm.js

```diff
@@ -195,8 +202,52 @@ async function checkPackageJson(filePath, opts) {
 	return changed;
 }
 
+/** @returns function to filter */
+function getFilterInLockFile(opts, exact) {
+	if (!exact) {
+		return key =>
+			(opts.scope && key.includes(`${opts.scope}/`)) ||
+			(opts.package && key.endsWith(`${opts.package}"`)) ||
+			(opts.startsWith && key.includes(opts.startsWith));
+	}
+	return key =>
+		(opts.scope && key.startsWith(`${opts.scope}/`)) ||
+		(opts.package && key === opts.package) ||
+		(opts.startsWith && key.startsWith(opts.startsWith));
+}
+
+async function removeFromLockFile(opts) {
+	let content;
+	try {
+		content = await fsprom.readFile(`${CWD}/package-lock.json`);
+	} catch (e) {
+		console.error(e);
+		return;
+	}
+	const pkgLock = JSON.parse(content);
+	console.log(pkgLock);
```

Comment:
is it ok to keep this log ?

Reply:
no, indeed I need to add tests, and find issues in the implementation, now it should be better